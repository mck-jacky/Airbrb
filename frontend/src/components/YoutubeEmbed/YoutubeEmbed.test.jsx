import React from 'react';
import { render, screen } from '@testing-library/react';
import YoutubeEmbed from './YoutubeEmbed'
import ShallowRenderer from 'react-test-renderer/shallow'

describe('YoutubeEmbed', () => {
  it('is in the document', () => {
    render(<YoutubeEmbed embedId='abcd' />)
    const component = screen.getByTestId('youtube-embed')
    expect(component).toBeInTheDocument();
  })

  it('has iframe', () => {
    render(<YoutubeEmbed embedId='abcd' />)
    const component = screen.getByTestId('youtube-embed')
    expect(component).toContainHTML('<iframe ')
  })

  it('has correct src', () => {
    render(<YoutubeEmbed embedId='abcd' />)
    const component = screen.getByTestId('youtube-embed')
    expect(component).toContainHTML('src="https://www.youtube.com/embed/abcd"')
  })

  it('has correct properties', () => {
    render(<YoutubeEmbed embedId='abcd' />)
    const component = screen.getByTestId('youtube-embed')
    expect(component).toContainHTML('width="100%')
    expect(component).toContainHTML('height="300')
    expect(component).toContainHTML('frameborder="0"')
    expect(component).toContainHTML('allowfullscreen=""')
    expect(component).toContainHTML('title="Embedded youtube"')
  })

  it('has to match the snapshot', () => {
    const renderer = ShallowRenderer.createRenderer()
    renderer.render(<YoutubeEmbed embedId='abcd' />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
