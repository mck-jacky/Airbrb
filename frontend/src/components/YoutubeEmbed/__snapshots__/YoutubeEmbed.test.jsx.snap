// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YoutubeEmbed has to match the snapshot 1`] = `
<div
  className="video-responsive"
  data-testid="youtube-embed"
>
  <iframe
    allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
    height="300"
    src="https://www.youtube.com/embed/abcd"
    title="Embedded youtube"
    width="100%"
  />
</div>
`;
